authentication/active_directory_authentication_provider.py

```diff
@@ -0,0 +1,60 @@
+#########
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+from ldap_authentication_provider import LDAPAuthenticationProvider
+
+
+DEFAULT_AUTH_ATTR_NAME = 'sAMAccountName'
+UPN_ATTR = 'userPrincipalName'
+
+
+class ActiveDirectoryAuthenticationProvider(LDAPAuthenticationProvider):
+
+    def __init__(self, domain_name=None, **kwargs):
+        super(ActiveDirectoryAuthenticationProvider, self).__init__(**kwargs)
+        self.domain_name = domain_name
+
+    # Active Directory authentication can be done by subtree match or according
+    # to the UPN attribute which is the default auth attr in Active Directory.
+    def authenticate(self, userstore=None):
+        who, password = self.get_creds_from_request()
+
+        if self.is_subtree_authentication_method():
+            self.authenticate_by_attribute(who,
+                                           password,
+                                           default_attr=DEFAULT_AUTH_ATTR_NAME)
```

Comment:
This method is actually used by both the ldap and the AD auth classes and the default values is used as a fallback in-case no specific user_id_attr was specified in the configuration. To clarify, LDAP's default attribute is uid while UPN is the default used by active directory